src/apis/httpClient.ts

```diff
@@ -0,0 +1,69 @@
+const getApiBaseUrl = () => {
+  if (import.meta.env.VITE_USE_MOCK === "true") {
+    return "https://techcourse-lv2-alb-974870821.ap-northeast-2.elb.amazonaws.com/";
+  }
+  return import.meta.env.VITE_API_BASE_URL;
+};
+
+export const API_BASE_URL = getApiBaseUrl();
+export const API_TOKEN = import.meta.env.VITE_TOKEN;
+
+class HTTPClient {
+  baseUrl = "";
+  apiKey = "";
+
+  constructor(baseUrl: string, apiKey?: string) {
+    this.baseUrl = baseUrl;
+    this.apiKey = apiKey || "";
+  }
+
+  private getHeaders() {
+    return {
+      "Content-Type": "application/json",
+      ...(this.apiKey && { Authorization: `Basic ${this.apiKey}` }),
+    };
+  }
+
```

Comment:
```suggestion
  private get headers() {
    return {
      "Content-Type": "application/json",
      ...(this.apiKey && { Authorization: `Basic ${this.apiKey}` }),
    };
  }
```
getter로 정의해주면 더 편할 것 같아요!